### PR TITLE
Making submissions persist to database and updating unit tests

### PR DESCRIPTION
**Pull request by abaumann**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@ case class WorkspaceName(
                       name: String) extends Identifiable {
   def path : String = "workspaces/" + namespace + "/" + name
   override def toString = namespace + "/" + name // used in error messages
-}
+}h
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: what?

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala [hunk 2]
```diff
@@ -54,38 +54,37 @@ class SubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpe
...
-  it should "400 when given no entity expression and an entity of the wrong type" in withWorkspaceService { workspaceService =>
-    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", "PairSet", "ps1", None)
-    val rqComplete = workspaceService.createSubmission( testData.wsName, submissionRq ).asInstanceOf[RequestComplete[(StatusCode, String)]]
-    val (status, _) = rqComplete.response
-    assertResult(StatusCodes.BadRequest) {
-      status
-    }
-  }
 
+  /* SUCCESS cases */
```
Comment (helgridly) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala]: Ugh, you monster. These were totally written in the same order as the branches in the createSubmission code.

Comment (abaumann) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala]: they should have followed the "with" order in createSubmission:
```scala
txn withGraph { graph =>
        withWorkspace(workspaceName, txn) { workspace =>
          requireAccess(GCSAccessLevel.Write, workspace, txn) {
            withMethodConfig(workspace, submission.methodConfigurationNamespace, submission.methodConfigurationName, txn) { methodConfig =>
              withEntity(workspace, submission.entityType, submission.entityName, txn) { entity =>
                withMethod(workspace, methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion, userInfo.authCookie) { agoraEntity =>
                  withWdl(agoraEntity) { wdl =>
                     withEVERYTHING { everyeverything =>
```

Comment (helgridly) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala]: You're right, and I don't know why they didn't. I'm still not a huge fan of splitting them into "success" and "fail" cases but I have bigger battles to fight than this one.

Comment (abaumann) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala]: I also dont care enough and will gladly remove those comments

**Comment by mattsooknah** [comment]
:+1: looks good enough to me

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 3]
```diff
@@ -337,11 +343,18 @@ trait GraphDAO {
 
     val parameters = ctor.asMethod.paramLists.head.map { paramSymbol =>
       val paramName = paramSymbol.name.decodedName.toString.trim
+
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: its kind of ugly that tryLoad is defined inside a map, but i didnt want to have to pass paramName, classT and vertex to that exception.  the other option was implicit arguments, but those would require paramName, classT and vertex to be implicit

**Pull request merged**